src/Stream/BidirectionalReader.h

```diff
@@ -6,12 +6,28 @@
 namespace OP2Utility::Stream
 {
 	class BidirectionalReader : public ForwardReader {
-	public:
-		inline void Peek(void* buffer, std::size_t size) {
+	protected:
+		// Generic peek method, which raises an exception if the full data cannot be read
+		// Note: This is named separately from Peek to prevent name hiding in derived classes
+		// All Peek methods are syntax sugar which delegate to this method
+		void PeekImplementation(void* buffer, std::size_t size)
+		{
 			ReadImplementation(buffer, size);
 			SeekBackward(size);
 		}
```

Comment:
Err, I don't think the name hiding concern is relevant to `Peek`, as it delegates to `ReadImplementation`, which does the class specific portion. There should be no need for derived classes to override `Peek` (or `PeekImplementation`). (It's not declared `virtual`).

Sorry, I probably wasn't being very clear on that point. It's probably fine to have a `public` low level `Peek` implementation, with the actual full implementation, and there should be no need to declare it `virtual`.